src/core/segment_buffers/implementations/audio_video/audio_video_segment_buffer.ts

```diff
@@ -421,76 +402,82 @@ export default class AudioVideoSegmentBuffer
       return; // still processing `this._pendingTask`
     }
 
-    // handle end of previous task if needed
     if (this._pendingTask !== null) {
-      if (this._pendingTask.type !== SegmentBufferOperation.Push ||
-          this._pendingTask.steps.length === 0)
-      {
-        switch (this._pendingTask.type) {
+      const task = this._pendingTask;
+      if (task.type !== SegmentBufferOperation.Push || task.data.length === 0) {
+        // If we're here, we've finished processing the task
+        switch (task.type) {
           case SegmentBufferOperation.Push:
-            if (this._pendingTask.inventoryData !== null) {
-              this._segmentInventory.insertChunk(this._pendingTask.inventoryData);
+            if (task.inventoryData !== null) {
+              this._segmentInventory.insertChunk(task.inventoryData);
             }
             break;
           case SegmentBufferOperation.EndOfSegment:
-            this._segmentInventory.completeSegment(this._pendingTask.value);
+            this._segmentInventory.completeSegment(task.value);
             break;
           case SegmentBufferOperation.Remove:
             this.synchronizeInventory();
             break;
           default:
-            assertUnreachable(this._pendingTask);
+            assertUnreachable(task);
         }
-        const { subject } = this._pendingTask;
+
+        const { subject } = task;
         this._pendingTask = null;
         subject.next();
         subject.complete();
-        if (this._queue.length > 0) {
-          this._flush();
-        }
+        this._flush(); // Go to next item in queue
         return;
       }
-    } else if (this._queue.length === 0) {
-      return; // we have nothing left to do
-    } else {
-      const newQueueItem = this._queue.shift();
-      if (newQueueItem === undefined) {
-        // TODO TypeScrypt do not get the previous length check. Find solution /
-        // open issue
-        throw new Error("An item from the AudioVideoSegmentBuffer queue was not defined");
-      }
-      this._pendingTask = convertQueueItemToTask(newQueueItem);
-      if (this._pendingTask === null) { // nothing to do, complete and go to next item
-        newQueueItem.subject.next();
-        newQueueItem.subject.complete();
-        this._flush();
-        return;
+    } else { // if this._pendingTask is null, go to next item in queue
+      const nextItem = this._queue.shift();
+      if (nextItem === undefined) {
+        return; // we have nothing left to do
+      } else if (nextItem.type !== SegmentBufferOperation.Push) {
+        this._pendingTask = nextItem;
+      } else {
+        const itemValue = nextItem.value;
+
+        let dataToPush : BufferSource[];
+        try {
```

Comment:
I'm not sure to see what could possibly go wrong here

Reply:
Me neither, but as we're using browser APIs everywhere in it, it was added as a safe measure.

For example, I'm not sure that mutating appendWindows or timestampOffset can break